.github/workflows/label-pull-requests.yml

```diff
@@ -0,0 +1,76 @@
+name: Pull request labeler
+
+# FIXME: Consider adding the `issue_comment` event to change labels based on generic, non-review pull request comments.
+#        Consider the trade off of how spammy it can be (one CI run per comment) and how useful it would be to have.
+#        Consider alternatives to `issue_comment`.
+on:
+  pull_request:
+    types: [opened, reopened, converted_to_draft, ready_for_review, synchronize, edited, review_requested]
+  pull_request_review:
+    types: [submitted, dismissed]
+
+jobs:
+  label_pull_request:
+    runs-on: ubuntu-22.04
+    if: always() && github.repository == 'SerenityOS/serenity'
+    
+    steps:
+      - name: Label pull request
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.BUGGIEBOT }}
```

Comment:
Are we sure the `BUGGIEBOT` token has permissions to manage labels?

Reply:
I'm not sure if the BUGGIEBOT token has sufficient permissions. For this to work, the token requires:
- Pull request write permissions to be able to add labels
- The ability to see the maintainers team and its members to be able to apply the `pr-maintainer-approved-but-awaiting-ci`
The other option is to use the GitHub Actions token which allows you to change the permissions on it via the YML.

Reply:
@IdanHo Do you recall what permissions the BUGGIEBOT token has, I think you created it?